### ISSUE TITLE
OCR API that takes an array of images, returns first found element 

### ISSUE THREAD

**Issue by mstan** [nut-tree/nut.js#186]
**Short overview**
Looking to pass an array of images and need one matched. Doesn't matter which

**Use case**
Needing to match any in a list of images. Which one that's matched is unimportant. 

**Detailed feature description**
Takes an array of images and returns a single region of some matched element. 

Consider setting it up in such a way that array order determines priority (step through array until there's a match?)

**Comment by s1hofmann** [comment]
Hi @mstan 👋

> Needing to match any in a list of images. Which one that's matched is unimportant.

Sounds like [Promise.any](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/any) solves your problem?

You could pass an iterable of multiple calls to `screen.find` and `Promise.any` would resolve to the first resolved Promise:

```ts
const { screen, mouse, straightTo, centerOf } = require("@nut-tree/nut-js");

(async () => {
    const firstMatch = await Promise.any([screen.find("one.png"), screen.find("two.png")]);
    await mouse.move(straightTo(centerOf(firstMatch)));
})();
```

**Comment by mstan** [comment]
Hey there!

I think this would actually address my concerns. Hadn't considered it .Will play with it. 

I'll re-open if I find a need. Will close out for now.

**Comment by mstan** [comment]
Hiya! Following up on this issue. Keeping it closed since I did get to a solution. But I wanted to follow up that my search criteria was particularly large (12+ images). A Promise.any appears to be executing this in parallel and will continue to execute even after an image is found for all the remaining failures. It was causing a _lot_ of lag for my use case.

I decided to try and tackle this with a sequential processing in a for loop. This way it will run one at a time and return if something if found, else error out.

```
	let location;
	for(let i = 0; i < fileNameArray.length; i++ ) {
		try {
			location = await this.findImage(fileNameArray[i]);
			break;
		} catch(error) {
			debug(error);
			continue;
		}
	}
	if(location) {
		return location;
	} else {
		throw new Error('No matches were found.');
	}
```

JavaScript is pretty quick and dirty. Likely not following best practices, but it gets the idea across.
